.github/workflows/build-push-containers.yml

```diff
@@ -56,18 +57,25 @@ jobs:
             type=semver,pattern={{version}}
       -
         name: Build and push dev image
+        id: build-dev
         uses: docker/build-push-action@v5
         with:
+          context: .
           push: true
           tags: ${{ steps.meta-dev.outputs.tags }}
           labels: ${{ steps.meta-dev.outputs.labels }}
           file: docker/dev.Dockerfile
           cache-from: type=gha
           cache-to: type=gha,mode=max
+
+      - name: Update prod image Dockerfile with dev image tag
+        run: |
+          sed -i "s/omnigibson-dev:latest/omnigibson-dev@${{ steps.build-dev.outputs.digest }}/g" docker/prod.Dockerfile && cat docker/prod.Dockerfile
```

Comment:
For reviewer context: this makes each branch's prod build use the same branch's dev build.